app/src/androidTest/java/ch/hikemate/app/authentication/FirebaseAuthRepositoryTest.kt

```diff
@@ -161,22 +166,51 @@ class FirebaseAuthRepositoryTest {
         coEvery { mockCredentialManager.getCredential(any(), any<GetCredentialRequest>()) } throws
             testException
 
-        val onSuccess: (FirebaseUser?) -> Unit = mockk(relaxed = true)
-        val onError: (Throwable) -> Unit = mockk(relaxed = true)
-
         repository.signInWithGoogle(
-            onSuccess = onSuccess,
-            onErrorAction = onError,
+            onSuccess = mockOnSuccess,
+            onErrorAction = mockOnError,
             context = context,
             credentialManager = mockCredentialManager,
-            coroutineScope = this)
+            coroutineScope = this,
+            startAddAccountIntentLauncher = null)
 
         advanceUntilIdle()
 
         coVerify { mockCredentialManager.getCredential(any(), any<GetCredentialRequest>()) }
         verify(exactly = 0) { mockFirebaseAuth.signInWithCredential(mockAuthCredential) }
-        verify(exactly = 0) { onSuccess(any()) }
-        verify { onError(any<GetCredentialUnknownException>()) }
+        verify(exactly = 0) { mockOnSuccess(any()) }
+        verify { mockOnError(any<GetCredentialUnknownException>()) }
+      }
+
+  @Test
+  fun testSignInWithGoogle_NoCredentialException() =
+      runTest(timeout = 5.seconds) {
+        coEvery { mockCredentialManager.getCredential(any(), any<GetCredentialRequest>()) } throws
+            NoCredentialException()
+
+        val mockLauncher: ManagedActivityResultLauncher<Intent, ActivityResult> =
+            mockk(relaxed = true)
+
+        // Call signInWithGoogle, expecting it to handle the NoCredentialException
+        repository.signInWithGoogle(
+            onSuccess = mockOnSuccess,
+            onErrorAction = mockOnError,
+            context = context,
+            credentialManager = mockCredentialManager,
+            coroutineScope = this,
+            startAddAccountIntentLauncher = mockLauncher)
+
+        advanceUntilIdle()
+
+        // Verify that getCredential() was called and threw the NoCredentialException
+        coVerify { mockCredentialManager.getCredential(any(), any<GetCredentialRequest>()) }
```

Comment:
Nitpick: This check only checks if getCredential() is called, but not for exception throwing

Reply:
You're right, and its not supposed to check for the throwing. I amended the comment.